src/validation/pou.rs

```diff
@@ -34,16 +40,65 @@ fn validate_pou<T: AnnotationMap>(validator: &mut Validator, pou: &Pou, context:
     if pou.pou_type == PouType::Function {
         validate_function(validator, pou, context);
     };
+    if pou.pou_type == PouType::Class {
+        validate_class(validator, pou, context);
+    };
+    if pou.pou_type == PouType::Program {
+        validate_program(validator, pou);
+    }
+}
+
+fn validate_class<T: AnnotationMap>(validator: &mut Validator, pou: &Pou, context: &ValidationContext<T>) {
+    // var in/out/inout blocks are not allowed inside of class declaration
+    if pou.variable_blocks.iter().any(|it| {
```

Comment:
You don't  need to do this but this might be simpler to right: 

```rust
any(|it|matches!(it.variable_block_type, VariableBlockType::InOut|VariableBlockType::Input(_)|
```